client/components/TestDisplay.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+class TestDisplay extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {tests:[], testName: ''};
+  }
+
+  componentDidMount() {
+    //load all test data when page loads
+    fetch('/api/getall')
+    .then((res) => {
+      return res.json();
+    })
+    .then((res) => {
+      this.setState({tests:res});
+
+      //classNames - required in code snippet
+      const classNames = res.map((test) => {
+        return 'openab-' + test.id;
+      })
+      this.setState({classNames:classNames});
+      console.log(this.state.classNames);
+      this.render();
```

Comment:
this.render() not necessary as this.setState() in makeSnippet() will always cause re-render [this.setState() will always call render unless there is logic in setState callback or in shouldComponentUpdate]